### PR TITLE
Update flake8 and black configs (Closes #55)

### LINKED ISSUE #55: Review Flake8 configuration, remove rules from ignore that would improve our code quality
While auditing the project code quality (#33), I noticed that several Flake8 rules were being ignored:

https://github.com/CiviWiki/OpenCiviWiki/blob/3c287290e907c0e1aa2c82f6588afcdb4b3b81e4/setup.cfg#L1-L2

While it may be desirable to ignore some Flake8 rules, there are likely rules being ignored that would encourage good coding practices. For example, we are lacking a lot of docstrings in CiviWiki modules and classes.

# Rules we currently ignore

## Continuation lines
- E121, continuation line under-indented for hanging indent
- E122, continuation line missing indentation or outdented
- E126, continuation line over-indented for hanging indent
- E127, continuation line over-indented for visual indent
- E128, continuation line under-indented for visual indent
  - https://stackoverflow.com/questions/15435811/what-is-pep8s-e128-continuation-line-under-indented-for-visual-indent
- E131, continuation line unaligned for hanging indent

## Comments
- E265, block comment should start with ‘# ‘
- E261, at least two spaces before inline comment
- E262, inline comment should start with ‘# ‘
- E266, too many leading ‘#’ for block comment

## Whitespace
- E501, line too long (82 > 79 characters)
- E302, expected 2 blank lines, found 0
- E303, too many blank lines (3)
- E226, missing whitespace around arithmetic operator
- E231, missing whitespace after ‘,’, ‘;’, or ‘:’
- E201, whitespace after ‘(‘
- E202, whitespace before ‘)’
- E203, whitespace before ‘:’
- W293, blank line contains whitespace
- W391, blank line at end of file
- W292, no newline at end of file
- W503, line break occurred before a binary operator

## Other
- E123, closing bracket does not match indentation of opening bracket’s line
- F403, ‘from module import *’ used; unable to detect undefined names
- E401, multiple imports on one line
- E731, do not assign a lambda expression, use a def

# Goal
Refine the Flake8 configuration. Ignore fewer Flake8 rules. Improve our code quality/practices.

# Tasl
- [ ] Add description of each ignored Flakd8 rule to this isse
- [ ] Determine which Flake8 rules to continue ignoring
- [ ] Remove useful Flake8 rules from the ignore list, to improve our project code quality

### PR DESCRIPTION
**Pull request by gorkemarslan**
Closes #55

- I make `flake8` compatible with `black` configs as I discussed in the related [issue](https://github.com/CiviWiki/OpenCiviWiki/issues/55#issuecomment-944929669).
- I add `black` configs. Unfortunately, `black` only accepts `pyproject.toml` file as a config file, not `setup.cfg`.
- Since Django migrations violate the `black` rules because of single-double quotes, I exclude `migrations` files for `black`.
- I move `setup.cfg `and `pyproject.toml` files to up one level, i.e. they are not located in `project` directory anymore.

@brylie and @AbhijithGanesh, do you have any suggestion to add it to `black` configs like excluding specific directories or rules?

### CODE REVIEW

**Comment by AbhijithGanesh** [comment]
I'll see and let you know Gorkem

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[tool.black]
+exclude = '''
+/(
+  | migrations
+)/
+'''
```
Comment (brylie) [inline pyproject.toml]: Add newline
```suggestion
'''

```

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+exclude = */migrations/*, */staticfiles/*, */env/*, */venv/*,
+
+[coverage:run]
+omit =
+    */env/*,
+    */venv/*,
+    *tests*,
+    */migrations/*,
+    manage.py,
+    *__init__.py,
```
Comment (brylie) [inline setup.cfg]: Add newline
```suggestion
    *__init__.py,

```

**Review by brylie — CHANGES_REQUESTED** [review]

**Comment by brylie** [comment]
> do you have any suggestion to add it to black configs like excluding specific directories or rules?

I think we should format all code that we are responsible for maintaining with black. So, only auto-generated and third-party code should be ignored. Ignoring `migrations` seems to be sufficient for now.

**Comment by gorkemarslan** [comment]
Screen Shot 2021-10-17 at 14 12 44

@brylie, if you want, I can format the above with `black` and commit here.

There is no `flake8` issue in the project. I guess all of the warnings are related to being single quotes.

**Comment by brylie** [comment]
OK, sounds good.

**Comment by gorkemarslan** [comment]
@brylie, from now on, there is no `black` and `flake8` warning.

**Review by brylie — APPROVED** [review]

**Pull request merged**